File: DSA/patterns/frequency_counter/MajorityElement.js
// MajorityElement.js

function findMajorityElement(arr) {
  /**
   * Find the majority element (element that appears more than n/2 times) using the Frequency Counter pattern.
   */
  let freqCount = {};

  for (let num of arr) {
    freqCount[num] = (freqCount[num] || 0) + 1;
    if (freqCount[num] > Math.floor(arr.length / 2)) {
      return num;
    }
  }

  return null;
}

File: DSA/patterns/frequency_counter/majority_element.py
"""majority_element.py"""

def find_majority_element(arr):
    """
    Find the majority element (element that appears more than n/2 times).
    using the Frequency Counter pattern.
    """
    freq_count = {}

    for num in arr:
        freq_count[num] = freq_count.get(num, 0) + 1
        if freq_count[num] > len(arr) // 2:
            return num

    return None
